templates/mail/htmlmail.php

```diff
@@ -0,0 +1,35 @@
+<table cellspacing="0" cellpadding="0" border="0" width="100%">
+<tr><td>
+<table cellspacing="0" cellpadding="0" border="0" width="600px">
+<tr>
+<td bgcolor="<?php p($theme->getMailHeaderColor());?>" width="20px">&nbsp;</td>
+<td bgcolor="<?php p($theme->getMailHeaderColor());?>">
+<img src="<?php p(\OC::$server->getURLGenerator()->getAbsoluteURL(image_path('', 'logo-mail.gif'))); ?>" alt="<?php p($theme->getName()); ?>"/>
+</td>
+</tr>
+<tr><td colspan="2">&nbsp;</td></tr>
+<tr>
+<td width="20px">&nbsp;</td>
+<td style="font-weight:normal; font-size:0.8em; line-height:1.2em; font-family:verdana,'arial',sans;">
+<?php p($_['subject']); ?>
+<br /><br />
+<?php p($_['message']); ?>
+<br /><br />
+<?php print_unescaped($l->t('Go to <a href="%s">%s</a> to check the notification', [$_['serverUrl'], $_['serverUrl']])); ?>
```

Comment:
each and every print_unescaped needs a close look so that no xss is possible

Reply:
The serverUrl variable is generated in https://github.com/owncloud/notifications/blob/master/lib/Mailer/NotificationMailerAdapter.php#L88 .
I guess it depends on the level of trust we have on the link provided by the notification or the one we're generating. We can harden the check to verify it's an url.

The other thing is the translation.

Reply:
I'm a bit late to this, but shouldn't the templates always auto-escape template variables?

Reply:
depends if it's escaped earlier or not. you don't want to double escape as it can make strings look odd